datalad/core/distributed/push.py

```diff
@@ -113,9 +113,10 @@ class Push(Interface):
             args=("--since",),
             constraints=EnsureStr() | EnsureNone(),
             doc="""specifies commit-ish (tag, shasum, etc.) from which to look for
-            changes to decide whether pushing is necessary.
-            If '^' is given, the last state of the current branch at the sibling
-            is taken as a starting point."""),
+            changes to decide whether pushing of data or recursion into subdatasets
+            (in recursive mode) is necessary.
+            If '^' is given, changes are looked up from the commit of the current
+            branch at the sibling."""),
```

Comment:
Hmm, this makes the situation less clear to me.  With the previous version, I would have assumed it was talking about the remote-tracking branch.  With this version, it sounds like the state on the remote may actually be considered.  Perhaps this could be clarified with a "last known state" somewhere in there.